Create `ScrollToTop` button to scroll on top of the page used in QuestinnaireEditor

AdityaJ2305 opened the pull request:
## Proposed Changes

- Fixes #12671 


https://github.com/user-attachments/assets/18bbc610-4d16-4b79-a7bc-8645b91cfdbb





@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Added a "Scroll to Top" button in the Questionnaire Editor that appears after scrolling down, enabling users to quickly return to the top of the page.  
  - Included localization support for the button’s label and tooltip.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AdityaJ2305 committed:
scroll to top

AdityaJ2305 committed:
Merge branch 'develop' into scroll_to_top

AdityaJ2305 commented:
Hey Team, 

Should I hide it for mobile ? little confused there , here is the screen recording for mobile:

https://github.com/user-attachments/assets/93bc3d47-9459-41ca-9eac-80805484875c

AdityaJ2305 committed:
coderabbit suggestion

AdityaJ2305 committed:
Merge branch 'develop' into scroll_to_top

AdityaJ2305 committed:
Merge branch 'develop' into scroll_to_top

Jacobjeevan commented on src/components/Common/ScrollToTop.tsx:
```diff
@@ -0,0 +1,36 @@
+  return (
+    <Button
+      onClick={scrollToTop}
+      className="fixed bottom-10 right-20 z-50 rounded-full shadow-md"
```
hmm, adjust the position for mobile, closer to the right side of the screen.

For desktop, it's fine for questionnaire editor as is since its width is constrained, but may not work for other pages.

AdityaJ2305 replied:
Updated to take smRight prop which is default to sm:right-20 and for mobile adjust the right side to right-2

AdityaJ2305 committed:
Merge remote-tracking branch 'upstream/develop' into scroll_to_top

AdityaJ2305 committed:
take right from prop

AdityaJ2305 committed:
take right prop

Jacobjeevan commented on src/components/Common/ScrollToTop.tsx:
```diff
@@ -6,11 +6,7 @@ import { cn } from "@/lib/utils";
-}: {
-  smRight?: string;
-}) {
+export function ScrollToTopButton({ right = 20 }: { right?: number }) {
```
works, but I'd suggest renaming this prop, not clear.

AdityaJ2305 replied:
rightOffset ?

AdityaJ2305 committed:
rightOffset

AdityaJ2305 commented:
@nihal467 
no, didn't discuss with vinu regarding this , I'll message in channel asking vinu to review the ui 👍

AdityaJ2305 committed:
update css

AdityaJ2305 committed:
button update

AdityaJ2305 committed:
implement vinu suggestion

AdityaJ2305 commented on src/components/Common/ScrollToTop.tsx:
```diff
@@ -0,0 +1,43 @@
+    <Button
+      onClick={scrollToTop}
+      className="fixed z-50 rounded-full shadow-md  hover:border-primary-700 hover:text-primary-700 hover:bg-primary-100"
+      style={{ right: rightOffset, bottom: bottomOffset }}
```
reason for using style:  https://stackoverflow.com/questions/72560190/using-dynamic-position-values-in-tailwind-css

noticed flakiness without it

rithviknishad replied:
all this hacks are not really needed in the first place. you can simply make the parent element specify the position for this button, using tailwind's classes itself.

AdityaJ2305 replied:
made parent specify the position by passing className as prop

AdityaJ2305 committed:
Merge branch 'develop' into scroll_to_top

AdityaJ2305 committed:
add className as prop

AdityaJ2305 committed:
shift css to parent for position
- now scrolltotop is just a button posisiton and everything is dependent on parent

AdityaJ2305 committed:
rm space

AdityaJ2305 committed:
Merge branch 'develop' into scroll_to_top

AdityaJ2305 committed:
minor css

nihal467 commented:
LGTM

rithviknishad commented on src/components/Common/ScrollToTop.tsx:
```diff
@@ -0,0 +1,41 @@
+  }, []);
+
+  const scrollToTop = () => {
+    window.scrollTo({ top: 0, behavior: "smooth" });
```
let css styling take care of it (which already is)

```suggestion
    window.scrollTo({ top: 0 });
```

rithviknishad committed:
Update src/components/Common/ScrollToTop.tsx
